carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/IDriverPool.java

```diff
@@ -87,7 +87,13 @@ default public WebDriver getDriver(String name, DesiredCapabilities capabilities
 
         ConcurrentHashMap<String, CarinaDriver> currentDrivers = getDrivers();
         if (currentDrivers.containsKey(name)) {
-            drv = currentDrivers.get(name).getDriver();
+        	CarinaDriver cdrv = currentDrivers.get(name);
+            drv = cdrv.getDriver();
+            if (Phase.BEFORE_SUITE.equals(cdrv.getPhase())) {
+            	logger.info("Before suite registered driver will be returned.");
+            } else {
+            	logger.debug(cdrv.getPhase() + " registered driver will be returned.");
+            } 
```

Comment:
What about adding these loggers constantly. It will help to determine possible issue during driver creation.